sdk/python/tests/unit/permissions/auth/client/test_authentication_client_manager_factory.py

```diff
@@ -0,0 +1,70 @@
+import os
+from unittest import mock
+
+import assertpy
+import jwt
+import pytest
+
+from feast.permissions.auth.auth_type import AuthType
+from feast.permissions.auth_model import (
+    KubernetesAuthConfig,
+    NoAuthConfig,
+    OidcClientAuthConfig,
+)
+from feast.permissions.client.auth_client_manager import (
+    AuthenticationClientManagerFactory,
+)
+from feast.permissions.client.intra_comm_authentication_client_manager import (
+    IntraCommAuthClientManager,
+)
+
+
+@mock.patch.dict(os.environ, {"INTRA_COMMUNICATION_BASE64": "server_intra_com_val"})
+@pytest.mark.parametrize(
+    "auth_config",
+    [
+        NoAuthConfig(),
+        KubernetesAuthConfig(**{"type": "kubernetes"}),
+        OidcClientAuthConfig(
+            **{
+                "type": "oidc",
+                "client_id": "feast-integration-client",
+                "client_secret": "feast-integration-client-secret",
+                "username": "reader_writer",
+                "password": "password",
+                "auth_discovery_url": "KEYCLOAK_URL_PLACE_HOLDER/realms/master/.well-known/openid-configuration",
+            }
+        ),
+    ],
+)
```

Comment:
why can't we use the default auth_config under root directory of `conftest.py`?

Reply:
To create the AuthConfig object at the beginning of the test we needed an ugly if else if else

Reply:
I see we are using if else to assert the different conditions based on different auth_config. Not sure if i misunderstood any thing.